### PR TITLE
New package: gtk+4-3.94.0

### PR DESCRIPTION
**Pull request by Cogitri**

### CODE REVIEW

File: srcpkgs/gtk+4/template [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Template file for 'gtk+4'
+pkgname=gtk+4
+version=3.94.0
+revision=1
+wrksrc="gtk+-${version}"
+build_style=meson
+build_helper="gir"
+configure_args="-Dman-pages=true
+ -Dbuild-tests=false
+ -Dmedia=all
```
Comment (Cogitri) [inline srcpkgs/gtk+4/template]: This links against both ffmpeg and gstreamer, not quite so sure if we want that.

File: srcpkgs/graphene-devel [hunk 2]
```diff
@@ -0,0 +1 @@
+graphene/
```
Comment (maxice8) [inline srcpkgs/graphene-devel]: no leading slash

File: srcpkgs/gtk+4-demo [hunk 3]
```diff
@@ -0,0 +1 @@
+gtk+4/
```
Comment (maxice8) [inline srcpkgs/gtk+4-demo]: no leading slash

File: srcpkgs/gtk+4/update [hunk 4]
```diff
@@ -0,0 +1,2 @@
+pkgname="gtk+"
```
Comment (maxice8) [inline srcpkgs/gtk+4/update]: gtk4 will be named without the + sign

File: srcpkgs/gtk+4/template [hunk 5]
```diff
@@ -0,0 +1,74 @@
+# Template file for 'gtk+4'
+pkgname=gtk+4
```
Comment (maxice8) [inline srcpkgs/gtk+4/template]: gtk4 won't have the plus sign

File: srcpkgs/gtk4/template [hunk 6]
```diff
@@ -0,0 +1,73 @@
+# Template file for 'gtk4'
+pkgname=gtk4
+version=3.94.0
+revision=1
+wrksrc="gtk+-${version}"
+build_style=meson
+build_helper="gir"
+configure_args="-Dman-pages=true
+ -Dbuild-tests=false
+ -Dmedia=all
```
Comment (maxice8) [inline srcpkgs/gtk4/template]: this can be put in the line above

**Comment by maxice8** [comment]
graphene is installing this:

```
=> graphene-1.8.2_1: running post-install hook: 06-strip-and-debug-pkgs ...
   Stripped position-independent executable: /usr/libexec/installed-tests/graphene-1.0/box
   Stripped position-independent executable: /usr/libexec/installed-tests/graphene-1.0/point3d
   Stripped position-independent executable: /usr/libexec/installed-tests/graphene-1.0/frustum
   Stripped position-independent executable: /usr/libexec/installed-tests/graphene-1.0/vec3
   Stripped position-independent executable: /usr/libexec/installed-tests/graphene-1.0/plane
   Stripped position-independent executable: /usr/libexec/installed-tests/graphene-1.0/size
   Stripped position-independent executable: /usr/libexec/installed-tests/graphene-1.0/euler
   Stripped position-independent executable: /usr/libexec/installed-tests/graphene-1.0/ray
   Stripped position-independent executable: /usr/libexec/installed-tests/graphene-1.0/rect
   Stripped position-independent executable: /usr/libexec/installed-tests/graphene-1.0/point
   Stripped position-independent executable: /usr/libexec/installed-tests/graphene-1.0/quaternion
   Stripped position-independent executable: /usr/libexec/installed-tests/graphene-1.0/sphere
   Stripped position-independent executable: /usr/libexec/installed-tests/graphene-1.0/quad
   Stripped position-independent executable: /usr/libexec/installed-tests/graphene-1.0/matrix
   Stripped position-independent executable: /usr/libexec/installed-tests/graphene-1.0/vec2
   Stripped position-independent executable: /usr/libexec/installed-tests/graphene-1.0/triangle
   Stripped position-independent executable: /usr/libexec/installed-tests/graphene-1.0/vec4
   Stripped position-independent executable: /usr/libexec/installed-tests/graphene-1.0/simd
```

Please check if there is a switch that can be disabled or remove them with rm -rf

**Comment by Cogitri** [comment]
Oh, okie, thought it'd be alright to install them  in either the normal or the devel package.

**Comment by maxice8** [comment]
```
/usr/libexec/installed-tests/graphene-1.0/box
/usr/libexec/installed-tests/graphene-1.0/euler
/usr/libexec/installed-tests/graphene-1.0/frustum
/usr/libexec/installed-tests/graphene-1.0/matrix
/usr/libexec/installed-tests/graphene-1.0/plane
/usr/libexec/installed-tests/graphene-1.0/point
/usr/libexec/installed-tests/graphene-1.0/point3d
/usr/libexec/installed-tests/graphene-1.0/quad
/usr/libexec/installed-tests/graphene-1.0/quaternion
/usr/libexec/installed-tests/graphene-1.0/ray
/usr/libexec/installed-tests/graphene-1.0/rect
/usr/libexec/installed-tests/graphene-1.0/simd
/usr/libexec/installed-tests/graphene-1.0/size
/usr/libexec/installed-tests/graphene-1.0/sphere
/usr/libexec/installed-tests/graphene-1.0/triangle
/usr/libexec/installed-tests/graphene-1.0/vec2
/usr/libexec/installed-tests/graphene-1.0/vec3
/usr/libexec/installed-tests/graphene-1.0/vec4
/usr/share/installed-tests/graphene-1.0/box.test
/usr/share/installed-tests/graphene-1.0/euler.test
/usr/share/installed-tests/graphene-1.0/frustum.test
/usr/share/installed-tests/graphene-1.0/matrix.test
/usr/share/installed-tests/graphene-1.0/plane.test
/usr/share/installed-tests/graphene-1.0/point.test
/usr/share/installed-tests/graphene-1.0/point3d.test
/usr/share/installed-tests/graphene-1.0/quad.test
/usr/share/installed-tests/graphene-1.0/quaternion.test
/usr/share/installed-tests/graphene-1.0/ray.test
/usr/share/installed-tests/graphene-1.0/rect.test
/usr/share/installed-tests/graphene-1.0/simd.test
/usr/share/installed-tests/graphene-1.0/size.test
/usr/share/installed-tests/graphene-1.0/sphere.test
/usr/share/installed-tests/graphene-1.0/triangle.test
/usr/share/installed-tests/graphene-1.0/vec2.test
/usr/share/installed-tests/graphene-1.0/vec3.test
/usr/share/installed-tests/graphene-1.0/vec4.test
```

We don't care about them

**Comment by maxice8** [comment]
They have their own update-icon-cache equivalent

/usr/bin/gtk4-update-icon-cache

**Comment by Cogitri** [comment]
Updated.

File: srcpkgs/gtk4/template [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+		vmove usr/share/gtk-4.0/gtkbuilder.rng
+		vmove usr/share/glib-2.0/schemas/org.gtk.Demo.gschema.xml
+		vmove usr/share/applications/gtk4-widget-factory.desktop
+		vmove usr/share/applications/gtk4-demo.desktop
+		vmove usr/share/icons
+	}
+}
+
+gtk4-update-icon-cache_package() {
+	short_desc+=" - The GTK4 update icon cache tool"
```
Comment (maxice8) [inline srcpkgs/gtk4/template]: no need for article 'The'

Comment (Cogitri) [reply srcpkgs/gtk4/template]: Fixed.

**Comment by Cogitri** [comment]
Anything left to do for this?

**Comment by maxice8** [comment]
> https://github.com/void-linux/void-packages/pull/8803#issuecomment-464724798

Problably need to add hooks/triggers for gtk4 components but i'm not sure

**Comment by Cogitri** [comment]
Rebased.

**Comment by maxice8** [comment]
Fails to compile:

```
@gdk-broadway@sta/broadway-server.c.o' -c ../gdk/broadway/broadway-server.c
../gdk/broadway/broadway-server.c:1162:10: fatal error: broadwayjs.h: No such file or directory
 #include "broadwayjs.h"
          ^~~~~~~~~~~~~~
```

**Comment by Cogitri** [comment]
On what platform? It works fine locally and ln Travis, it seems.

**Comment by maxice8** [comment]
On Wed, Mar 13, 2019 at 10:44:23PM -0700, Rasmus Thomsen wrote:
> On what platform? It works fine locally and ln Travis, it seems.
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/8803#issuecomment-472712359

x86_64

**Comment by Cogitri** [comment]
Hum, passed again on Travis, could you try again?

**Comment by Cogitri** [comment]
Just tested locally again, works just fine. Could you maybe send the do_configure output? (or the entire build log for that matter)

**Comment by Gottox** [comment]
I'd like to get gtk4 merged early. This is against our "no beta software" policy, but this way we can stabilize gtk4 in the tree and integrate it faster once the final is released. Objections?

**Comment by Cogitri** [comment]
Sounds good, but the merge commit seems fishy

**Comment by Gottox** [comment]
@Cogitri yea, may rebase yourself?

**Comment by Cogitri** [comment]
Done

**Pull request merged**